shared/react-native/android/app/src/main/java/io/keybase/ossifrage/MainActivity.java

```diff
@@ -59,6 +74,102 @@ private void createDummyFile() {
         }
     }
 
+    private ReactContext getReactContext() {
+        ReactInstanceManager instanceManager = getReactInstanceManager();
+        if (instanceManager == null) return null;
+
+        return instanceManager.getCurrentReactContext();
+    }
+
+    private void handleNotificationIntent(Intent intent) {
+        Bundle bundle = intent.getExtras();
+        if (bundle == null || !bundle.containsKey("notification")) return;
+
+        ReactContext reactContext = getReactContext();
+        if (reactContext == null) return;
+
+        DeviceEventManagerModule.RCTDeviceEventEmitter emitter = reactContext
+                .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class);
+        if (emitter != null) {
+            emitter.emit("androidIntentNotification", "");
+        }
+    }
+
+    private void handleSendIntentStream(Intent intent) {
+        Uri uri = intent.getParcelableExtra(Intent.EXTRA_STREAM);
+        if (uri == null) return;
+
+        String filePath = null;
+        if (uri.getScheme().equals("content")) {
+            ContentResolver resolver = getContentResolver();
+            String mimeType = resolver.getType(uri);
+            String extension = MimeTypeMap.getSingleton().getExtensionFromMimeType(mimeType);
+            String filename = String.format("%s.%s", UUID.randomUUID().toString(), extension);
+            File file = new File(getCacheDir(), filename);
+            try {
+                InputStream istream = resolver.openInputStream(uri);
+                OutputStream ostream = new FileOutputStream(file);
+
+                byte[] buf = new byte[64 * 1024];
+                int len;
+                while ((len = istream.read(buf)) != -1) {
+                    ostream.write(buf, 0, len);
+                }
+                filePath = file.getPath();
+            } catch (IOException ex) {
+                Log.w(TAG, "error writing shared file " + uri.toString());
+            }
```

Comment:
What do you think if we use this routine to handle onShareText so we always get a file that's actionable for us?

Reply:
I was thinking of doing that, but it won't apply when we want to share to chat. Still figuring out the best approach for that.